src/Common/Core/Impl/Diagnostics/Check.cs

```diff
@@ -19,16 +22,25 @@ public static class Check {
             }
         }
 
+        [DebuggerStepThrough]
         public static void ArgumentOutOfRange(string argumentName, Func<bool> predicate) {
             if (predicate()) {
                 throw new ArgumentOutOfRangeException(argumentName);
             }
         }
 
+        [DebuggerStepThrough]
         public static void InvalidOperation(Func<bool> predicate, string message = null) {
             if (!predicate()) {
                 throw new InvalidOperationException(message ?? string.Empty);
             }
         }
+
+        [DebuggerStepThrough]
+        public static void Argument(string argumentName, Func<bool> predicate) {
+            if (!predicate()) {
+                throw new ArgumentException();
```

Comment:
`ArgumentException($"{argumentName} is not valid", argumentName)` or something like that